pkg/workceptor/kubernetes.go

```diff
@@ -127,8 +128,16 @@ func (kw *kubeUnit) createPod(env map[string]string) error {
 			return fmt.Errorf("At least one container must be named worker")
 		}
 		spec.RestartPolicy = corev1.RestartPolicyNever
+		userNamespace := pod.ObjectMeta.Namespace
+		if userNamespace != "" {
+			ked.KubeNamespace = userNamespace
+		}
+		userPodName := pod.ObjectMeta.Name
+		if userPodName != "" {
+			kw.namePrefix = userPodName + "-"
```

Comment:
I am not sure about this. If a `name` is passed as part of the Pod spec, I taking that and using it for `GenerateName`.